[Huang Qiming] iP

ceilingFan456 opened the pull request:
### Paimon

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :+1: 

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]   Managing deadlines (coming soon)
- [ ]   Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ceilingFan456 committed:
Rename Duke to Paimon

ceilingFan456 committed:
Add ability to repeat words.

ceilingFan456 committed:
Add item list to keep track of user inputs.

ceilingFan456 committed:
Add feature to mark and unmark tasks.

ceilingFan456 committed:
Add Todo, Deadline, Event classes.

ceilingFan456 committed:
Add text-ui-test scripts

ceilingFan456 committed:
Handle user invalid inputs with exceptions.

ceilingFan456 committed:
Add delete function.

ceilingFan456 committed:
Update gitignore

ceilingFan456 committed:
Done with saving items to file and loading at the start

ceilingFan456 committed:
Merge branch 'branch-Level-7'

ceilingFan456 committed:
Done with adding date time parser

ceilingFan456 committed:
Merge branch 'Level-8'

ceilingFan456 committed:
Have items folder

ceilingFan456 committed:
Storage and Tasklist in progress

ceilingFan456 committed:
Done with Storage and Tasklist

ceilingFan456 committed:
Parser, Command classes in progress

ceilingFan456 committed:
Done and tested OOP components.

ceilingFan456 committed:
Merge branch 'branch-A-MoreOOP'

ceilingFan456 committed:
Done with making paimon a package.

ceilingFan456 committed:
Merge branch 'branch-A-Packages'

ceilingFan456 committed:
Merge branch 'add-gradle-support'

ceilingFan456 committed:
Add gradle support

ceilingFan456 committed:
Add tests for Parser and TaskList classes. Add equal methods for
commands and items

ceilingFan456 committed:
Merge branch 'branch-A-JUnit'

ceilingFan456 committed:
Add jar file creation

ceilingFan456 committed:
Merge branch 'branch-A-Jar'

ceilingFan456 committed:
Check coding styles

ceilingFan456 committed:
Add javadoc for all public methods

ceilingFan456 committed:
Add find command

ceilingFan456 committed:
Merge branch 'branch-A-JavaDoc'

ceilingFan456 committed:
Merge branch 'branch-A-CodingStandard'

ceilingFan456 committed:
Merge branch 'branch-Level-9'

ceilingFan456 committed:
Add checkstyle scripts

ceilingFan456 committed:
Update build.gradle

ceilingFan456 committed:
Remove trailing white space check

ceilingFan456 committed:
Add checkstyle and correct all formats

farhan-navas commented on src/main/java/paimon/ui/UI.java:
```diff
@@ -0,0 +1,50 @@
+ * UI class that interacts with the user.
+ * For example, it reads input from users. 
+ */
+public class UI {
```
```suggestion
public class Ui {
```
Class name should be Ui and not UI: [[Relevant link in guide] ](https://se-education.org/guides/conventions/java/intermediate.html#naming) under the abbreviations section.

ceilingFan456 replied:
Thank you!

farhan-navas commented on src/main/java/paimon/commands/Command.java:
```diff
@@ -0,0 +1,30 @@
+    /**
+     * Execute the command with current tasklist and ui
+     * 
+     * @param t tasklist involved
```
@param should end with a full stop
```suggestion
     * @param t tasklist involved.
```

farhan-navas reviewed (commented):
Overall LGTM, very small coding standard nits to fix.

shawnnygoh commented on src/main/java/paimon/items/Todo.java:
```diff
@@ -0,0 +1,64 @@
+     * 
+     * @return string with standard format to represent the item.
+     */
+    public String to_save() {
```
Should to_save() be in snake case or camelCase?
```suggestion
    public String toSave() {
```

shawnnygoh commented on src/main/java/paimon/tasklist/TaskList.java:
```diff
@@ -0,0 +1,102 @@
+    /**
+     * Lists all the current items in the list and outputs to screen. 
+     */
+    public void list_items() {
```
Should list_items() be in snake case or camelCase?
```suggestion
    public void listItems() {
```

shawnnygoh commented on src/test/java/paimon/tasklist/TaskListTest.java:
```diff
@@ -0,0 +1,63 @@
+
+public class TaskListTest {
+    @Test
+    public void addTest(){
```
Does it make more sense to have addTest() or testAdd() as the method name here?
```suggestion
    public void testAdd(){
```

shawnnygoh commented on src/test/java/paimon/tasklist/TaskListTest.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    @Test
+    public void sizeTest(){
```
Does it make more sense to have sizeTest() or testSize() as the method name here?
```suggestion
    public void testSize(){
```

shawnnygoh commented on src/test/java/paimon/tasklist/TaskListTest.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    @Test
+    public void markTest(){
```
Does it make more sense to have markTest() or testMark() as the method name here?
```suggestion
    public void testMark(){
```

shawnnygoh reviewed (commented):
Looks good. 👍🏻
Your code mostly follows the code quality standards for naming.
Good job!

ceilingFan456 committed:
Update src/main/java/paimon/ui/UI.java

Co-authored-by: Farhan Navas <farhanmnavas@gmail.com>

ceilingFan456 committed:
Update src/main/java/paimon/commands/Command.java

Co-authored-by: Farhan Navas <farhanmnavas@gmail.com>

ceilingFan456 committed:
Save for later

ceilingFan456 committed:
Merge branch 'master' into branch-B-Level-10

ceilingFan456 committed:
Save for later

ceilingFan456 committed:
Fix format issue

ceilingFan456 committed:
Finish part 3

FooNicholas commented on src/main/java/paimon/parser/Parser.java:
```diff
@@ -0,0 +1,81 @@
+     */
+    public static Command parse(String str) {
+        try {
+            if (str.equals("bye")) {
```
A switch statement could be used here possibly as an alternative.

FooNicholas commented on src/main/java/paimon/items/Event.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    @Override
+    public String to_save() {
```
Should to_save be in camelCase or snake case?

FooNicholas commented on src/main/java/paimon/items/Todo.java:
```diff
@@ -0,0 +1,64 @@
+package paimon.items;
+
+public class Todo {
```
A possible alternative design would be to design a Task interface / abstract class which the 3 types of task extend from, for scenarios where a Task that does not share the attributes / methods for Todo is required to be added.

FooNicholas commented on src/main/java/paimon/Paimon.java:
```diff
@@ -0,0 +1,50 @@
+import paimon.ui.UI;
+
+/** 
+ * Paimon is a chatbot that keeps track of tasks for users. 
```
Good job on including the Javadoc comments for all classes.

FooNicholas commented on src/main/java/paimon/Paimon.java:
```diff
@@ -0,0 +1,50 @@
+     */
+    public void run() {
+        this.ui.greet();
+        boolean running = true;
```
Should this Boolean value be named as isRunning instead?

FooNicholas reviewed (commented):
LGTM.  Provided some possible alternative suggestions.

ceilingFan456 committed:
Save for later

ceilingFan456 committed:
Correct UI to Ui

ceilingFan456 committed:
Add assert statements

ceilingFan456 committed:
Merge branch 'branch-A-Assertions'

ceilingFan456 committed:
Merge branch 'master' into branch-B-Level-10

ceilingFan456 committed:
Add GUI

ceilingFan456 committed:
Merge branch 'branch-B-Level-10'

ceilingFan456 committed:
Correct mistakes in arrowhead style and slap

ceilingFan456 committed:
Merge branch 'branch-A-CodeQuality'

ceilingFan456 committed:
Add Undo function

ceilingFan456 committed:
Merge branch 'branch-A-BCD-Extension'

ceilingFan456 committed:
Add function to change paimon images

ceilingFan456 committed:
Add random paimon images

ceilingFan456 committed:
Merge branch 'branch-A-BetterGui'

ceilingFan456 committed:
Fix format issue

ceilingFan456 committed:
Add user guide

ceilingFan456 committed:
Merge branch 'branch-A-UserGuide'

ceilingFan456 committed:
Change format

ceilingFan456 committed:
Handle invalid datetime format gracefully

ceilingFan456 committed:
Add welcome message

Currently there is no display of messages when user starts the
program.

Users may feel lost and do not know what to do next.

Add a welcome message function in MainWindow and Paimon class.

This ensures there is always a welcome message display from MainWindow
class and gives more flexibility if the welcome message has to be
changed in the future.

(attempt to write full commit message)

ceilingFan456 committed:
MainWindow class: Bind vbox height to scrollPane

Currently after users type a msg, the chat does not
automatically scroll to the latest conversation.

This might be inconvenient for users as they need to keep
scrolling their conversations down.

A Listener has been added to the height property of VBox. Whenever
the height increases, the scrollPane will reset its height back
to 1.0

The height property of VBox has been bind to the scrollPane
viewport height property.

(attempt to write a full commit msg)

ceilingFan456 committed:
Remove trailing and leading whitespace in user commands

Currently commands like " list" or "list " are recognised as
invalid commands.

This is not ideal as users may be confused after following
the intructions to type the commands.

Trim in user input in Parse.parse function.

Adding it as part of the command validation code block ensures
all syntax checkings are at one place. It is also neater if
Parser class handles all the syntax checks before passing the strings
to the downstream classes.

ceilingFan456 committed:
Check empty command parameters

Currently if any any command parameter is empty, the program
quits with runtime error.

This should be handled explicitly and the program should remind
user the correct way of keying in the command parameters.

Throws PaimonInvalidInputException if the command parameter is empty.

This is done in each parsing function. Although this is a bit
repetitive, it is necessary as we do not long the length of
command and required parameters without first parsing them.

(attempt to write full commit msg)

ceilingFan456 committed:
Update README.md about undo command

ceilingFan456 committed:
Fix format and testcases

ceilingFan456 committed:
Merge branch 'master' into branch-A-CodeQuality

ceilingFan456 committed:
Add dummy comment for pull request practice

ceilingFan456 committed:
Merge branch 'master' into branch-A-Assertions

ceilingFan456 committed:
Add dummy msg for PR practice

ceilingFan456 committed:
Merge pull request #3 from ceilingFan456/branch-A-Assertions

Branch a assertions

ceilingFan456 committed:
Merge branch 'master' into branch-A-CodeQuality

ceilingFan456 committed:
Merge pull request #2 from ceilingFan456/branch-A-CodeQuality

Branch a code quality

ceilingFan456 committed:
Fix boolean variable naming issue
